usaspending_api/agency/v2/views/sub_agency.py

```diff
@@ -0,0 +1,138 @@
+                    "total_obligations": bucket.get("total_subagency_obligations").get("value"),
+                    "transaction_count": bucket.get("doc_count"),
+                    "new_award_count": bucket.get("agency_award_count").get("agency_award_value").get("value"),
+                    "children": sorted(
+                        self.format_child_response(bucket.get("offices").get("buckets")),
+                        key=lambda x: x.get(self.pagination.sort_key),
+                        reverse=self.pagination.sort_order == "desc",
+                    ),
+                }
+            )
+        return response
+
+    def format_child_response(self, children):
+        response = []
+        for child in children:
+            response.append(
+                {
+                    "name": child.get("office_info")
+                    .get("hits")
+                    .get("hits")[0]
+                    .get("_source")
+                    .get(f"{self.agency_type}_office_name"),
+                    "total_obligations": child.get("total_office_obligations").get("value"),
+                    "transaction_count": child.get("doc_count"),
+                    "new_award_count": child.get("office_award_count").get("office_award_value").get("value"),
+                }
+            )
+        return response
+
+    def get_sub_agency_list(self):
+        response = self.query_elasticsearch()
+        return self.format_elasticsearch_results(response)
+
+    def query_elasticsearch(self):
+        fiscal_year = FiscalYear(self.fiscal_year)
+        filter_query = QueryWithFilters.generate_transactions_elasticsearch_query(
+            {
+                "agencies": [{"type": self.agency_type, "tier": "toptier", "name": self.toptier_agency.name}],
+                "time_period": [{"start_date": fiscal_year.start.date(), "end_date": fiscal_year.end.date()}],
+                "award_type_codes": self._query_params.get("award_type_codes", []),
+            }
+        )
+        search = TransactionSearch().filter(filter_query)
+        subagency_dim_metadata = A(
+            "top_hits",
+            size=1,
+            _source={"includes": [f"{self.agency_type}_subtier_agency_abbreviation"]},
+        )
+        office_dim_metadata = A(
+            "top_hits",
+            size=1,
+            _source={"includes": [f"{self.agency_type}_office_name"]},
+        )
+        subtier_agency_agg = A("terms", field=f"{self.agency_type}_subtier_agency_name.keyword")
+        office_agg = A("terms", field=f"{self.agency_type}_office_code.keyword")
+        agency_obligation_agg = A("sum", field="generated_pragmatic_obligation")
+        office_obligation_agg = A("sum", field="generated_pragmatic_obligation")
+        new_award_filter = A(
+            "filter", range={"award_date_signed": {"gte": fiscal_year.start.date(), "lte": fiscal_year.end.date()}}
+        )
+        new_award_agg = A("cardinality", field="award_id")
```

Comment:
Are you certain that the value here won't ever be above 10k? Might be able to use a terms aggregation to get the count. The value of `doc_count_error_upper_bound` might not be 0, but since all Transactions from an award should be routed to the same shard we might be okay.

Reply:
Yeah it's possible that this could go over 10k, i switched it out for the scripted metric which should work above 10k